app/Models/GameSet.php

```diff
@@ -61,6 +62,31 @@ public static function boot()
         parent::boot();
 
         static::pivotAttached(function ($model, $relationName, $pivotIds, $pivotIdsAttributes) {
+            if ($relationName === 'viewRoles' || $relationName === 'updateRoles') {
+                /** @var User $user */
+                $user = Auth::user();
+
+                $attachedRoles = Role::whereIn('id', $pivotIds)
+                    ->select(['id', 'name'])
+                    ->get();
+
+                $permission = $relationName === 'viewRoles'
+                    ? GameSetRolePermission::View->value
+                    : GameSetRolePermission::Update->value;
+
+                activity()->causedBy($user)->performedOn($model)
+                    ->withProperty('old', [$relationName => null])
+                    ->withProperty('attributes', [$relationName => $attachedRoles
+                        ->map(fn ($role) => [
+                            'id' => $role->id,
+                            'name' => $role->name,
+                            'permission' => $permission,
+                        ]),
+                    ])
+                    ->event('pivotAttached')
+                    ->log('pivotAttached');
```

Comment:
Not sure if there's an easy way to consolidate these into a single activity. I added two create roles and one update role and it created three activity entries.
![image](https://github.com/user-attachments/assets/ff7f8126-f367-410e-97c6-7ea2c0aadebc)

Reply:
Indirectly related: My Add Role window always lists all available roles by ID, even if I add search text.
![image](https://github.com/user-attachments/assets/35f5f0bd-c463-4e45-a27f-731c0e563a34)

Reply:
> Not sure if there's an easy way to consolidate these into a single activity. I added two create roles and one update role and it created three activity entries.

This is really tough unless we move all the logging to explicitly live inside `HubResource`. I'm not sure this trade-off is worth it.

> Indirectly related: My Add Role window always lists all available roles by ID, even if I add search text.

I was able to get this working right in latest for the "Attach role" dialog. I couldn't get the filtering to work right in the select controls in `HubResource` no matter what I tried.